src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1542,6 +1542,19 @@
     <Field Name="ResultKind" PropertyOverrides="true" Type="LookupResultKind"/>
   </Node>
 
+  <!--
+  This node is used to represent a target-typed object creation expression
+  It does not survive past initial binding.
```

Comment:
nit: This only survives past initial binding in error cases

Reply:
actually it doesn't (due to BindToNaturalType usages), unless we visit unconverted nodes (like in NullableWalker)